provisioner/shell/provisioner.go

```diff
@@ -202,20 +210,38 @@ func (p *Provisioner) Provision(ctx context.Context, ui packersdk.Ui, comm packe
 		scripts = append(scripts, tf.Name())
 
 		// Write our contents to it
-		writer := bufio.NewWriter(tf)
-		_, _ = writer.WriteString(fmt.Sprintf("#!%s\n", p.config.InlineShebang))
+		fileWriter := bufio.NewWriter(tf)
+		commandBuffer := &bytes.Buffer{}
 		for _, command := range p.config.Inline {
 			p.config.ctx.Data = generatedData
 			command, err := interpolate.Render(command, &p.config.ctx)
 			if err != nil {
 				return fmt.Errorf("Error interpolating Inline: %s", err)
 			}
-			if _, err := writer.WriteString(command + "\n"); err != nil {
+			if _, err := commandBuffer.WriteString(command + "\n"); err != nil {
```

Comment:
Since we're iterating on this code, I would also suggest moving to use a `fmt.Fprintf` here so we don't do the extra concatenation needlessly.
```suggestion
			if _, err := fmt.Fprintf(commandBuffer, "%s\n", command); err != nil {
```